app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,27 @@
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(username: params[:session][:email]) || User.find_by(email: params[:session][:email])
```

Comment:
I forgot to delete that private method. We had issues with the strong params because we are hijacking the form_for :email tag to include either a username or email, and the strong params could not differentiate the two.